allspice/utils/list_components.py

```diff
@@ -340,16 +395,44 @@ def _fetch_generated_json(repo: Repository, file_path: str, ref: Ref) -> dict:
     raise TimeoutError(f"Failed to fetch JSON for {file_path} after 5 attempts.")
 
 
-def _extract_schdoc_list_from_prjpcb(prjpcb_ini: configparser.ConfigParser) -> set[str]:
+@functools.cache
+def _fetch_all_files_in_repo(repo: Repository) -> list[str]:
     """
-    Get a list of SchDoc files from a PrjPcb file.
+    Fetch a list of all files in a repository at the default branch.
     """
 
-    return {
-        section["DocumentPath"]
-        for (_, section) in prjpcb_ini.items()
-        if "DocumentPath" in section and section["DocumentPath"].endswith(".SchDoc")
-    }
+    tree = repo.get_tree(recursive=True)
+
+    return [file.path for file in tree]
+
+
+def _extract_schdoc_list_from_prjpcb(
+    prjpcb_ini: configparser.ConfigParser,
+) -> tuple[set[str], set[str]]:
+    """
+    Extract sets of all schematic files used in this project.
+
+    :param prjpcb_ini: The contents of the PrjPcb file as a ConfigParser.
+    :returns: two sets. The first set is of relative paths from the porject
+    file to all the Schematic Documents in this project, and the strings are
+    paths to the schematic documents from the project file, and the second is
+    of relative paths to device sheets from the project file.
+    """
+
+    schdoc_files = set()
+    device_sheets = set()
+
+    for section in prjpcb_ini.sections():
+        if section.casefold().startswith("document"):
+            document_path = prjpcb_ini[section].get("DocumentPath")
+            if document_path and document_path.endswith(".SchDoc"):
```

Comment:
```suggestion
            if document_path and document_path.lower().endswith(".schdoc"):
```
Should this be case-insensitive? I expect so.